### PR TITLE
add matching cap option

### LINKED ISSUE #1310: Add matching caps to QF calculation
**User Story:**
As a
I want to
So that

**Acceptance Criteria**
GIVEN
WHEN
THEN

**Product & Design Links:**

**Tech Details:**

**Open Questions:**

**Notes/Assumptions:**
- This includes writing test coverage for the functionality added in the ticket, if necessary

### PR DESCRIPTION
**Pull request by gravityblast**
closes https://github.com/gitcoinco/grants-stack/issues/1310 and #3

### CODE REVIEW

**Comment by vacekj** [comment]
Nice, LGTM. Also - could you please create the branches using the "Create branch" button in the issue, or at least append the issue number at the start of the branch name? Makes it much easier to manage the branch - PR - issue relationship. Thanks!

**Review by vacekj — APPROVED** [review]

**Comment by gravityblast** [comment]
@vacekj cool I'll do it!

**Review by hansmrtn — APPROVED** [review]

File: src/test/linear.test.ts [hunk 1]
```diff
@@ -217,16 +231,56 @@ describe("linearQF", () => {
...
+        capOverflow: -28.4,
+        sumOfSqrt: 8,
+        totalReceived: 10,
+        matchedWithoutCap: 21.6,
+        matched: 28.4,
+      });
+
+      expect(res["project_3"]).toEqual({
+        // we will change this test in the next PRs using BigInt and cents or token decimals
+        capOverflow: 14.799999999999997,
```
Comment (boudra) [inline src/test/linear.test.ts:283]: nice

Comment (gravityblast) [reply src/test/linear.test.ts:283]: @boudra te gusta el floating point lol?

Comment (hansmrtn) [reply src/test/linear.test.ts:283]: lol

**Review by thelostone-mc — APPROVED** [review]

**Pull request merged**